src/sonar_widget/SonarPlot.cc

```diff
@@ -206,6 +225,16 @@ void SonarPlot::sonarPaletteChanged(int index){
     applyColormap((ColorGradientType) index);
 }
 
+// update the sector scan (for scanning sonars)
+void SonarPlot::setSectorScan(bool continuous, base::Angle left, base::Angle right){
+    this->continuous = continuous;
+    this->leftLimit = left;
+    this->rightLimit = right;
+
+    if (!continuous)
+        changedSectorScan = true;
```

Comment:
The changedSectorScan variable is used to clean the accumulated sonar data in setData().